core/include/scipp/core/value_and_variance.h

```diff
@@ -127,6 +127,22 @@ constexpr auto operator/(const T1 a, const ValueAndVariance<T2> b) noexcept {
                                            (b.value * b.value) /
                                            (b.value * b.value)};
 }
+template <class T1, class T2>
+constexpr auto operator<(const ValueAndVariance<T1> a, const T2 b) noexcept {
+  return ValueAndVariance{a.value < b, a.variance};
+}
+template <class T1, class T2>
+constexpr auto operator>(const ValueAndVariance<T1> a, const T2 b) noexcept {
+  return ValueAndVariance{a.value > b, a.variance};
+}
+template <class T1, class T2>
+constexpr auto operator>=(const ValueAndVariance<T1> a, const T2 b) noexcept {
+  return ValueAndVariance{a.value >= b, a.variance};
+}
+template <class T1, class T2>
+constexpr auto operator<=(const ValueAndVariance<T1> a, const T2 b) noexcept {
+  return ValueAndVariance{a.value <= b, a.variance};
+}
```

Comment:
I think I would avoid adding these operators for `ValueAndVariance`. It is a bit ill-defined what they should do, I think, so making those operators accessible for all uses of `transform` is a bit dangerous. Instead, use a locally defined lambda/operator?

Reply:
I agree. Particularly around the variance.